.env.example

```diff
@@ -10,50 +10,47 @@ DB_SSL="true"
 SQLITE="NO"
 SQLITE_DB_PATH="./sql.db"
 
-# Name of DB table
-TABLE="your_db_table"
-
-# If you want to activate embedding media filenames found in the DB (e.g. for the Gallery or Map views), 
-# set the following value to YES and provide the path to the media directory
-EMBED_MEDIA="NO"
-MEDIA_BASE_PATH="./media"
-
-# Front end dropdown filter options
-FRONT_END_FILTERING="YES"
-FRONT_END_FILTER_FIELD="Category"
-
 # Port to serve app
 PORT="8080"
 
-# Password to access the app and generate JWT token
+# OPTION 1 FOR AUTHENTICATION: Password to access the app and generate JWT token
 USE_PASSWORD="NO"
 PASSWORD="guardian-connector"
 
-# Auth0 credentials
+# Secret JWT key that can be used to bypass logging in for embedding purposes
+SECRET_JWT_KEY="secret-jwt-key"
+
+# OPTION 2 FOR AUTHENTICATION: Auth0 credentials
 NUXT_ENV_AUTH0_DOMAIN="domain.us.auth0.com"
 NUXT_ENV_AUTH0_CLIENT_ID=""
 NUXT_ENV_AUTH0_CLIENT_SECRET=""
 NUXT_ENV_AUTH0_AUDIENCE="https://audience.auth0.com/api/v2/"
 NUXT_ENV_AUTH0_BASE_URL='http://localhost:3000'
 
-# Secret JWT key that can be used to bypass logging in for embedding purposes
-SECRET_JWT_KEY="secret-jwt-key"
-
-# Generate an API key to add request headers made by the Nuxt front end 
+# Generate an API key to add to request headers made by the Nuxt front end 
 VUE_APP_API_KEY=""
 
-# Mapbox (all values optional except access token)
+# Mapbox access token
 MAPBOX_ACCESS_TOKEN="your_mapbox_token"
-MAPBOX_STYLE="mapbox://styles/mapbox/streets-v12"
-MAPBOX_PROJECTION="globe"
-MAPBOX_CENTER_LATITUDE="-15"
-MAPBOX_CENTER_LONGITUDE="0"
-MAPBOX_ZOOM="2.5"
-MAPBOX_PITCH="0"
-MAPBOX_BEARING="0"
-MAPBOX_3D="NO"
-
-# Unwanted SQL colums to hide in views
-# Refer to /docs/schemas.md for reference for what columns to hide
-UNWANTED_COLUMNS="latitude,longitude,coordinates"
-UNWANTED_SUBSTRINGS="meta,version,formhub,xform,attachments,tags,notes,topic,sender,geolocation,submission_time,submitted_by,status,supplementalDetails,validation_status,start,end,location"
\ No newline at end of file
+
+# Multi-line config providing views configuration for each database table
+# See docs/config.md for a full explanation for these settings.
+NUXT_ENV_VIEWS_CONFIG = "{
```

Comment:
Is the stuff here inside NUXT_ENV_VIEWS_CONFIG likely to change from deployment to deployment?  If not (if, say, we believe one certain group of settings is optimal across the board) then an alternative is to bake this into the Docker image itself -- through a Javascript or JSON or even .env file (if Javascript's own parser can deal with multi-line config).

This would be in the vein of superset's `[superset_config.py](https://github.com/ConservationMetrics/superset-deployment/blob/main/docker/pythonpath/superset_config.py)` file. There we went a step further which is to read the few things that do need per-deployment overrides from their own environment variables.  This, too, gets clunky.

Or, your `dockerenv.sh` script is good too :+1:

Reply:
Unfortunately a good number of the variables inside NUXT_ENV_VIEWS_CONFIG (for example, the mapbox parameters) do change quite a bit from deployment to deployment. 

I did consider a run script that does the same as `dockerenv.sh` but feeds the output to Docker directly instead of baking a separate `.env.docker` file, but I thought there was value in having persisting access to the output too (e.g. for easy config of Azure or other cloud deployments).